src/solver/simulation/sim_calcul_economique.cpp

```diff
@@ -638,10 +631,12 @@ void SIM_RenseignementProblemeHebdo(const Study& study,
             if (problem.CaracteristiquesHydrauliques[k].PresenceDHydrauliqueModulable > 0)
             {
                 auto& area = *study.areas.byIndex[k];
-                uint tsIndex = (NumeroChroniquesTireesParPays[numSpace][k]).Hydraulique;
-                auto& inflowsmatrix = area.hydro.series->storage;
+                auto& hydroSeries = area.hydro.series;
+                uint tsIndex = hydroSeries->getIndex(year);
+
+                auto& inflowsmatrix = hydroSeries->storage;
                 auto const& srcinflows = inflowsmatrix[tsIndex < inflowsmatrix.width ? tsIndex : 0];
-                auto& mingenmatrix = area.hydro.series->mingen;
+                auto& mingenmatrix = hydroSeries->mingen;
                 auto const& srcmingen = mingenmatrix[tsIndex < mingenmatrix.width ? tsIndex : 0];
```

Comment:
We could have a getter for this column in the HydroSeries class, instead of going through 1. retrieve the index then 2. retrieve the column.